### PR TITLE
Implementation of GoogleCloudExtensions

### PR DESCRIPTION
**Pull request by seehamrun**
I added getCryptoKeyStore and getBucketStore to the CloudExtension since I suspect they might be needed (the CloudFactory had these). 

I'm not sure of where the initialization of the constructor params should go. My initial thought was to pull them in from the ExtensionContext, but its not clear to me what that looks like for things like creds or the json factories. 

#192

### CODE REVIEW

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudExtension.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.dataportabilityproject.api.launcher.ExtensionContext;
+import org.dataportabilityproject.spi.cloud.extension.CloudExtension;
+import org.dataportabilityproject.spi.cloud.storage.BucketStore;
+import org.dataportabilityproject.spi.cloud.storage.CryptoKeyStore;
+import org.dataportabilityproject.spi.cloud.storage.JobStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GoogleCloudExtension implements CloudExtension {
+  // TODO: Should these be initialized from the ExtensionContext and not injected?
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudExtension.java]: Yep, the context sounds right

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudExtension.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  private static final Logger logger = LoggerFactory.getLogger(GoogleCloudExtension.class);
+
+  private boolean initialized = false;
+  private Datastore datastore;
+  private JobStore jobstore;
+  private BucketStore bucketStore;
+  private GoogleCryptoKeyStore cryptoKeyManagementSystem;
+
+  @Inject
+  GoogleCloudExtension(
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudExtension.java]: I think extensions will have no constructor if I understand correctly?

Comment (rtannenbaum) [reply extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudExtension.java]: I think just a public no-arg one right? I think that's required by the ServiceLoader

Comment (seehamrun) [reply extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudExtension.java]: Changed to have no constructor. I'm not convinced I'm creating the transport and the json factory correctly, but thats how it was done throughout the project - I don't have anything runnable to test it with though

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudExtension.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+      // TODO: Hook these up with the global instances
+      transport =  GoogleNetHttpTransport.newTrustedTransport();
+      jsonFactory = new JacksonFactory();
+      cryptoKeyManagementSystem = new GoogleCryptoKeyStore(transport, jsonFactory, projectId);
+
+      initialized = true;
+    } catch (IOException | GoogleCredentialException | GeneralSecurityException e) {
+      // TODO: the method doesn't throw an exception, how do we pass this onto the user?
+      logger.debug("Error initializing extension: " + this.getClass().getName(), e);
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudExtension.java]: logger.warn

**Comment by seehamrun** [comment]
Oh good point Jim! The extension context is something that would get created in our Main class, correct? I think that works because at that point we should know what each extension requires. 

I've switched to pulling them from the ExtensionContext with .getService(); but havent written up the context yet.

**Pull request merged**